components/automate-ui/src/app/entities/layout/layout-sidebar.service.ts

```diff
@@ -1,294 +1,321 @@
-import { Injectable } from '@angular/core';
+import { Injectable, OnInit, OnDestroy } from '@angular/core';
 import { Store } from '@ngrx/store';
-import { Observable } from 'rxjs';
+import { Subject, Observable, BehaviorSubject } from 'rxjs';
+import { forEach } from 'lodash';
+// import { takeUntil } from 'rxjs/operators';
 
 import { NgrxStateAtom } from 'app/ngrx.reducers';
 import { isProductDeployed } from 'app/staticConfig';
 import { isIAMv2 } from 'app/entities/policies/policy.selectors';
 import { FeatureFlagsService } from 'app/services/feature-flags/feature-flags.service';
 import { clientRunsWorkflowEnabled } from 'app/entities/client-runs/client-runs.selectors';
 import * as fromClientRuns from 'app/entities/client-runs/client-runs.reducer';
-
-import { MenuItemGroup } from './layout.model';
+import { UpdateSidebars } from './layout.actions';
+import { Sidebars } from './layout.model';
 
 @Injectable({
     providedIn: 'root'
 })
-export class LayoutSidebarService {
+export class LayoutSidebarService implements OnInit, OnDestroy {
     public applicationsFeatureFlagOn: boolean;
     public chefInfraServerViewsFeatureFlagOn: boolean;
-    public isIAMv2$: Observable<boolean>;
-    public ServiceNowfeatureFlagOn: boolean;
-    private workflowEnabled: boolean;
-
+    public isIAMv2: Observable<boolean>;
```

Comment:
Please stay with the convention of always naming Observables with a trailing `$`. That convention is used throughout our code base and makes it easier to digest the code. (That's not just a Chef convention, but a general rxjs convention.)

Reply:
@msorens Yes, you're right! I had switched between an extracted value back to observable. I will correct it! Thanks 😊